c/certificateService.c

```diff
@@ -109,110 +91,86 @@ static void respondWithInvalidMethod(HttpResponse *response) {
   HttpRequest *request = response->request;
 
   if (!strcmp(request->method, methodPOST))
   {
-    RUsermapParamList *userMapCertificateStructure 
-      = (RUsermapParamList*)safeMalloc31(sizeof(RUsermapParamList),"RUsermapParamList");
-    memset(userMapCertificateStructure, 0, sizeof(RUsermapParamList));
 
-    if(request->contentLength > sizeof(userMapCertificateStructure->certificate) || request->contentLength < 0) {
-      respondWithBadRequest(response);
-      return 0;
+    int urlLength = strlen(request->uri);
+    if(urlLength < 0 || urlLength > 17) {
+        respondWithJsonError(response, "URI is longer than maximum length of 17 characters.", 400, "Bad Request");
+        return 0;
     }
 
-    userMapCertificateStructure->certificateLength = request->contentLength;
-    memset(userMapCertificateStructure->certificate, 0, request->contentLength);
-    memcpy(userMapCertificateStructure->certificate, request->contentBody, request->contentLength);
+    char translatedURL[urlLength + 1];
+    strcpy(translatedURL, request->uri);
+    a2e(translatedURL, sizeof(translatedURL));
+    char *x509URI = strstr(translatedURL, "x509");
+    char *dnURI = strstr(translatedURL, "dn");
 
-    userMapCertificateStructure->functionCode = MAP_CERTIFICATE_TO_USERNAME;
+    char useridRacf[9];
+    int returnCodeRacf = 0;
+    int reasonCodeRacf = 0;
     int rc;
-
-#ifdef _LP64 
-    __asm(ASM_PREFIX
-	  /* We get the routine pointer for IRRSIM00 by an, *ahem*, direct approach.
-	     These offsets are stable, and this avoids linker/pragma mojo */
-	  " LA 15,X'10' \n"
-          " LG 15,X'220'(,15) \n" /* CSRTABLE */
-          " LG 15,X'28'(,15) \n" /* Some RACF Routin Vector */
-          " LG 15,X'A0'(,15) \n" /* IRRSIM00 itself */
-	  " LG 1,%0 \n"
-	  " SAM31 \n"
-	  " BALR 14,15 \n"
-          " SAM64 \n"
-          " ST 15,%0"
-	  :
-	  :"m"(userMapCertificateStructure),"m"(rc)
-	  :"r14","r15");
-#else
-    rc = IRRSIM00(
-        &userMapCertificateStructure->workarea, // WORKAREA
-        &userMapCertificateStructure->safRcAlet  , // ALET
-        &userMapCertificateStructure->returnCode,
-        &userMapCertificateStructure->racfRcAlet,
-        &userMapCertificateStructure->returnCodeRacf,
-        &userMapCertificateStructure->racfReasonAlet,
-        &userMapCertificateStructure->reasonCodeRacf,
-        &userMapCertificateStructure->fcAlet,
-        &userMapCertificateStructure->functionCode,
-        &userMapCertificateStructure->optionWord,
-        &userMapCertificateStructure->useridLengthRacf,
-        &userMapCertificateStructure->certificateLength,
-        &userMapCertificateStructure->applicationIdLength,
-        &userMapCertificateStructure->distinguishedNameLength,
-        &userMapCertificateStructure->registryNameLength
-    );
-#endif
+    if(x509URI != NULL) {
+    //  Certificate to user mapping
+        if(request->contentLength > 4096 || request->contentLength < 1) {
```

Comment:
Is 4096 an arbitrary limit?

Reply:
it is maximum length of the certificate from R_usermap docs, since there is this check as a part of the rusermap service in common-c I have removed it.

Reply:
Ok, but FYI the function in zowe-common-c doesn't check negative values which are allowed by the type.

Reply:
I can add the check for negative values, I was not aware that it would be possible